src/Data/Number.js

```diff
@@ -0,0 +1,10 @@
+exports.fromStringImpl = function (just) {
+  return function (nothing) {
+    return function (string) {
+      var result = parseFloat(string);
+
+      return isNaN(result) ? nothing
```

Comment:
I believe `isNaN` is not enough here, because `parseFloat` can also return `Infinity` (for inputs like `1e1000`). `isFinite` would be a better choice, I think.

Thinking about this, it would be very cool to have `isNaN` and `isFinite` available in purescript-numbers. Then, `fromString` could also be implemented in PureScript.